syncer/ddl.go

```diff
@@ -151,13 +153,38 @@ func (s *Syncer) splitAndFilterDDL(
 			}
 		}
 
-		shouldSkip, err2 := s.skipQuery(tableNames, stmt2, sql)
+		shouldSkip, isBaList, err2 := s.skipQuery(tableNames, stmt2, sql)
 		if err2 != nil {
 			return nil, nil, err2
 		}
 		if shouldSkip {
 			metrics.SkipBinlogDurationHistogram.WithLabelValues("query", s.cfg.Name, s.cfg.SourceID).Observe(time.Since(ec.startTime).Seconds())
-			ec.tctx.L().Warn("skip event", zap.String("event", "query"), zap.String("statement", sql), zap.String("schema", schema))
+			ec.tctx.L().Warn("skip event", zap.String("event", "query"), zap.String("sql", sql), zap.String("schema", schema), zap.Any("tableNames", tableNames))
+
+			if s.cfg.ShardMode == config.ShardOptimistic || isBaList || ec.currentLocation == nil {
+				continue
+			}
+			// TODO: wrap to a new func?
+			targetTableNames := make([]*filter.Table, 0, len(tableNames))
+			for i := range tableNames {
+				renamedSchema, renamedTable := s.renameShardingSchema(tableNames[i].Schema, tableNames[i].Name)
+				tableName := &filter.Table{
+					Schema: renamedSchema,
+					Name:   renamedTable,
+				}
+				targetTableNames = append(targetTableNames, tableName)
+			}
+
+			if err = s.trackDDL(schema, sql, [][]*filter.Table{tableNames, targetTableNames}, stmt2, &ec); err != nil {
+				return nil, nil, err
+			}
+
+			ec.tctx.L().Info("save table checkpoint for source", zap.String("event", "query"), zap.String("source", ec.traceSource), zap.Stringer("start location", *ec.startLocation), log.WrapStringerField("end location", *ec.currentLocation))
+			s.saveGlobalPoint(*ec.currentLocation)
+
+			if err = s.flushCheckPoints(); err != nil {
```

Comment:
we may need to add a flush job, to wait existing jobs executed. Because we saved the global checkpoint at above line but the jobs before that checkpoint may not be executed